From 7072fd9fb47f937ebfd68e0b83d80b4748023118 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:39:38 -0400
Subject: [ethernet] i40e: Add debugfs hooks to print current total FD filter count

Message-id: <1409658042-22689-218-git-send-email-sassmann@redhat.com>
Patchwork-id: 90414
O-Subject: [RHEL7.1 PATCH 217/281] i40e: Add debugfs hooks to print current total FD filter count
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

"fd current cnt" can be used to print the total filters consumed
by this interface, this includes guaranteed and best effort filters.

Change-ID: I2c417810c4999ce1388d2ea26f8e69679ba33966
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit e17ff05c5d9ee64550030d03e63719e6dc62d729)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_debugfs.c b/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
index 910b01b..ec07332 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
@@ -1743,6 +1743,9 @@ static ssize_t i40e_dbg_command_write(struct file *filp,
   i40e_dbg_cmd_fd_ctrl(pf, I40E_FLAG_FD_ATR_ENABLED, false);
  } else if (strncmp(cmd_buf, "fd-atr on", 9) == 0) {
   i40e_dbg_cmd_fd_ctrl(pf, I40E_FLAG_FD_ATR_ENABLED, true);
+ } else if (strncmp(cmd_buf, "fd current cnt", 14) == 0) {
+  dev_info(&pf->pdev->dev, "FD current total filter count for this interface: %d\n",
+    i40e_get_current_fd_count(pf));
  } else if (strncmp(cmd_buf, "lldp", 4) == 0) {
   if (strncmp(&cmd_buf[5], "stop", 4) == 0) {
    int ret;
@@ -1962,6 +1965,7 @@ static ssize_t i40e_dbg_command_write(struct file *filp,
   dev_info(&pf->pdev->dev, "  rem fd_filter <dest q_index> <flex_off> <pctype> <dest_vsi> <dest_ctl> <fd_status> <cnt_index> <fd_id> <packet_len> <packet>\n");
   dev_info(&pf->pdev->dev, "  fd-atr off\n");
   dev_info(&pf->pdev->dev, "  fd-atr on\n");
+  dev_info(&pf->pdev->dev, "  fd current cnt");
   dev_info(&pf->pdev->dev, "  lldp start\n");
   dev_info(&pf->pdev->dev, "  lldp stop\n");
   dev_info(&pf->pdev->dev, "  lldp get local\n");
-- 
1.7.1